types/bluebird/index.d.ts

```diff
@@ -209,144 +152,7 @@ declare class Bluebird<R> implements PromiseLike<R>, Bluebird.Inspection<R> {
-    onReject: (error: E1 | E2 | E3 | E4) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-  caught<U, E1, E2, E3, E4>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    filter2: ((error: E2) => boolean) | (E2 & object),
-    filter3: ((error: E3) => boolean) | (E3 & object),
-    filter4: ((error: E4) => boolean) | (E4 & object),
-    onReject: (error: E1 | E2 | E3 | E4) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-
-  caught<E1 extends Error, E2 extends Error, E3 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    filter2: (new (...args: any[]) => E2),
-    filter3: (new (...args: any[]) => E3),
-    onReject: (error: E1 | E2 | E3) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<E1, E2, E3>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    filter2: ((error: E2) => boolean) | (E2 & object),
-    filter3: ((error: E3) => boolean) | (E3 & object),
-    onReject: (error: E1 | E2 | E3) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<U, E1 extends Error, E2 extends Error, E3 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    filter2: (new (...args: any[]) => E2),
-    filter3: (new (...args: any[]) => E3),
-    onReject: (error: E1 | E2 | E3) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-  caught<U, E1, E2, E3>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    filter2: ((error: E2) => boolean) | (E2 & object),
-    filter3: ((error: E3) => boolean) | (E3 & object),
-    onReject: (error: E1 | E2 | E3) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-
-  caught<E1 extends Error, E2 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    filter2: (new (...args: any[]) => E2),
-    onReject: (error: E1 | E2) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<E1, E2>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    filter2: ((error: E2) => boolean) | (E2 & object),
-    onReject: (error: E1 | E2) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<U, E1 extends Error, E2 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    filter2: (new (...args: any[]) => E2),
-    onReject: (error: E1 | E2) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-  caught<U, E1, E2>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    filter2: ((error: E2) => boolean) | (E2 & object),
-    onReject: (error: E1 | E2) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-
-  caught<E1 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    onReject: (error: E1) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<E1>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    onReject: (error: E1) => R | PromiseLike<R>,
-  ): Bluebird<R>;
-  caught<U, E1 extends Error>(
-    filter1: (new (...args: any[]) => E1),
-    onReject: (error: E1) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
-  caught<U, E1>(
-    filter1: ((error: E1) => boolean) | (E1 & object),
-    onReject: (error: E1) => U | PromiseLike<U>,
-  ): Bluebird<U | R>;
+  caught: Bluebird<R>["catch"];
```

Comment:
Dude, this is f**** brilliant!
Of course… And it's such an obvious solution even… Why didn't we use that all the time yet?!

![Mind Blown](https://i.imgur.com/WrKPhfd.gif)